### ISSUE TITLE
First heketi-cli node add succeeds. Subsequent node or device adds fail.

### ISSUE THREAD

**Issue by andrewlove** [heketi/heketi#690]
Using heketi v4.0
Create a new cluster using `heketi-cli cluster create`
Add a first storage node using `heketi-cli node add --zone=1 --cluster=adae8d4511669f04f83e08bb6d6bc666 --management-host-name=ose-gluster-001.openstacklocal --storage-host-name=10.10.10.11
So far so good.

Add a second storage node using `heketi-cli node add --zone=1 --cluster=adae8d4511669f04f83e08bb6d6bc666 --management-host-name=ose-gluster-002.openstacklocal --storage-host-name=10.10.10.12
Get an error:
`Error: No pods with the label 'glusterfs-node=ose-gluster-001.openstacklocal' were found`

Add a device to the first storage node using `heketi-cli device add --node=c0a37eb19c729da6bd861de6d89ea5ba --name=/dev/vdb`
Get the same error as above:
`Error: No pods with the label 'glusterfs-node=ose-gluster-001.openstacklocal' were found`

Basically any heketi-cli management I try from this point gives me the same error:
`Error: No pods with the label 'glusterfs-node=ose-gluster-001.openstacklocal' were found`

**Comment by andrewlove** [comment]
Update
---------
I have gotten round the above error. I believe I may have not set the `HEKETI_KUBE_NAMESPACE` correctly, hence the PeerProbe errors.  

When i try to add a second storage node I now get:  
`Error: Unable to execute command on glusterfs-dc-ose-gluster-001.openstacklocal-1-m649t: peer probe: failed: Probe returned with Transport endpoint is not connected`

**Comment by andrewlove** [comment]
Update
---------
Having done an:  
`iptables -F`  
On each storage node I can now execute heketi-cli commands successfully.

The storage nodes/containers were created in OpenShift using the Heketi glusterfs-template.json

How do I ensure that there are no iptable rules in future gluster container deployments?

**Comment by lpabon** [comment]
@andrewlove good question.  Iptables is needed to be done by the kubernetes administrator because it is change on the host and not managed by OpenShift/Kubernetes.  You may want to use an ansible script on the nodes to setup the iptables accordingly.

**Comment by andrewlove** [comment]
@lpabon many thanks. I should just clarify, I issued `iptables -F` in each of the glusterfs containers not on the hosts.
As you say I manage the iptables on the hosts externally .
So my question is where do the Openshift/Kube containers pick up their iptables rules from?

**Comment by lpabon** [comment]
@andrewlove That is a great question.  I have no idea also :-) .  I thought you ran that on the host, didn't know you ran that from the container.  I wonder if the iptables is global for a privileged container (as the glusterfs one is), so what you did could be done and thus affects the host.  If that is true, then I am guessing that a non-privileged container would not allow an iptables change.
